pythonforandroid/graph.py

```diff
@@ -99,17 +129,125 @@ def find_order(graph):
+
+            # See if what was added before conflicts with this recipe:
+            for dep_tuple_list in deps:
+                if len(dep_tuple_list) > 1:
+                    # Not an obvious commitment, so we won't check.
+                    # (remember this function only catches obvious issues)
+                    continue
+                try:
+                    conflicts = [c.lower() for c in
+                                 Recipe.get_recipe(dep_tuple_list[0], ctx).
+                                 conflicts]
+                    if name in conflicts:
+                        triggered_conflicts.append(dep_tuple_list)
+                except ValueError:
+                    pass
+
+            # Throw error on conflict:
+            if len(triggered_conflicts) > 0:
+                # Get first conflict and see who added that one:
+                adder_recipe_2_name = "'||'".join(triggered_conflicts[0])
+                adder_recipe_2_name =\
+                    (deps_were_added_by.get(adder_recipe_2_name,
+                                            adder_recipe_2_name) or
+                     adder_recipe_2_name)  # might be in dict, but None
+                # Prompt error:
+                raise BuildInterruptingException(
+                    "Conflict detected:" +
+                    " '" + str(adder_recipe_1_name) + "'" +
+                    " inducing dependencies " + str((recipe.name,)) +
+                    " and '" + str(adder_recipe_2_name) + "'" +
+                    " inducing conflicting dependencies " +
+                    str(triggered_conflicts[0])
+                )
+
+            # Actually add it to our list:
+            deps.add(added_tuple)
+            deps_were_added_by[added_tuple] = adding_recipe
+
+            # Schedule dependencies to be added
+            to_be_added += [
+                (dep, adder_recipe_1_name or name)
+                for dep in recipe_dependencies
+                if dep not in deps
+            ]
+    # If we came here, then there were no obvious conflicts.
+    return None
+
+
+def get_recipe_order_and_bootstrap(ctx, names, bs=None, blacklist=[]):
+    # Get set of recipe/dependency names, clean up and add bootstrap deps:
+    names = set(names)
     if bs is not None and bs.recipe_depends:
-        recipes_to_load = recipes_to_load.union(set(bs.recipe_depends))
+        names = names.union(set(bs.recipe_depends))
+    names = fix_deplist([
+        ([name] if not isinstance(name, (list, tuple)) else name)
+        for name in names
+    ])
+    blacklist = [bitem.lower() for bitem in blacklist]
 
-    possible_orders = []
+    # Remove all values that are in the blacklist:
+    names_before_blacklist = list(names)
+    names = []
+    for name in names_before_blacklist:
+        cleaned_up_tuple = tuple([
+            item for item in name if item not in blacklist
+        ])
+        if len(cleaned_up_tuple) > 0:
```

Comment:
Minor: here and other place let's be Pythonic, for once the PEP8 allows us to be implicit, let's be :wink: 
```python
if not cleaned_up_tuple:
```

Reply:
fixed, thanks :+1: